src/benchmarklib/benchmark_runner.cpp

```diff
@@ -339,7 +343,9 @@ void BenchmarkRunner::_benchmark_ordered() {
       std::cout << "  -> Waiting for clients that are still running\n" << std::flush;
     }
     Hyrise::get().scheduler()->wait_for_all_tasks();
+
     Assert(_currently_running_clients == 0, "All runs must be finished at this point.");
+    deplete_semaphore(_running_clients_semaphore);
```

Comment:
What's the benefit over calling `_running_clients_semaphore.acquire()` directly? This should also just spin, right?

Reply:
First, `acquire()` blocks and as we don't know the internal count, we cannot know how often we shall call it.

Second it, it does spin for a tiny duration and then uses system semaphores to sleep.